### PR TITLE
Azure AD Reporting quarterly updates Q4 FY2021

### PR DESCRIPTION
**Pull request by besiler**
Added properties to the signin resource:

- authenticationRequirementPolicies
- autonomousSystemNumber
- crossTenantAccessType
- flaggedForReview
- homeTenantId
- isInteractive
- isTenantRestricted
- privateLinkDetails
- resourceTenantId
- alternateSignInName
- servicePrincipalCredentialKeyId
- servicePrincipalCredentialThumbprint
- signInEventTypes
-

### CODE REVIEW

File: api-reference/beta/resources/signin.md [hunk 1]
```diff
@@ -34,35 +34,51 @@ Provides details about user or application sign-in activity in your directory. Y
...
+|homeTenantId|String|The tenant identifier of the user initiating the sign-in. Not applicable in Managed Identity or Service Principal sign-ins.|
 |id|String|The identifier representing the sign-in activity. Supports `$filter` (`eq` operator only).|
 |ipAddress|String|The IP address of the client from where the sign-in occurred. Supports `$filter` (`eq` and `startsWith` operators only).|
-|isInteractive|Boolean|Indicates whether a sign-in is interactive or not.|
+|isInteractive|Boolean|Indicates whether a user sign-in is interactive or not. In interactive sign-ins, the user provides an authentication factor to Azure AD. These factors include passwords, responses to MFA challenges, biometric factors, or QR codes that a user provides to Azure AD or to an associated app. In non-interactive sign-ins, the user does not provide an authentication factor; instead, the client app uses a token or code to authenticate or access a resource on behalf of a user. Non-interactive sign ins are commonly used for a client to sign in on a user's behalf in a process transparent to the user.|
+|isTenantRestricted|Boolean|**TODO: Add Description**|
```
Comment (dkershaw10) [inline api-reference/beta/resources/signin.md:50]: Make sure you fix the TODOs

File: api-reference/beta/api/signin-get.md [hunk 2]
```diff
@@ -110,16 +110,32 @@ Content-type: application/json
   "appId": "de8bc8b5-5555-6666-a8ad-b748da725064",
   "appDisplayName": "Graph explorer",
   "authenticationRequirement": "MultifactorAuthentication",
+  "authenticationRequirementPolicies":
```
Comment (dkershaw10) [inline api-reference/beta/api/signin-get.md:113]: This doesn't look correct.  The property table says this is a collection.
Either this should have a value or not.  If not then it should be null or an empty array - [],

File: api-reference/beta/resources/signin.md [hunk 3]
```diff
@@ -34,35 +34,51 @@ Provides details about user or application sign-in activity in your directory. Y
...
 |deviceDetail|[deviceDetail](devicedetail.md)|The device information from where the sign-in occurred. Includes information such as deviceId, OS, and browser. Supports `$filter` (`eq` and `startsWith` operators only) on **browser** and **operatingSytem** properties.|
+|flaggedForReview|Boolean|During a failed sign in, a user may click a button in the Azure AD UX to mark the failed event for tenant admins. If a user clicked the button to flag a sign-in event, this value is true.|
+|homeTenantId|String|The tenant identifier of the user initiating the sign-in. Not applicable in Managed Identity or Service Principal sign-ins.|
 |id|String|The identifier representing the sign-in activity. Supports `$filter` (`eq` operator only).|
 |ipAddress|String|The IP address of the client from where the sign-in occurred. Supports `$filter` (`eq` and `startsWith` operators only).|
-|isInteractive|Boolean|Indicates whether a sign-in is interactive or not.|
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/signin.md:44]: missing property: **ipAddressFromResourceProvider**.

File: api-reference/beta/resources/signin.md [hunk 4]
```diff
@@ -34,35 +34,51 @@ Provides details about user or application sign-in activity in your directory. Y
...
+|isInteractive|Boolean|Indicates whether a user sign-in is interactive or not. In interactive sign-ins, the user provides an authentication factor to Azure AD. These factors include passwords, responses to MFA challenges, biometric factors, or QR codes that a user provides to Azure AD or to an associated app. In non-interactive sign-ins, the user does not provide an authentication factor; instead, the client app uses a token or code to authenticate or access a resource on behalf of a user. Non-interactive sign ins are commonly used for a client to sign in on a user's behalf in a process transparent to the user.|
+|isTenantRestricted|Boolean|**TODO: Add Description**|
 |location|[signInLocation](signinlocation.md)|The city, state, and 2 letter country code from where the sign-in occurred. Supports `$filter` (`eq` and `startsWith` operators only) on **city**, **state**, and **countryOrRegion** properties.|
+|mfaDetail|[mfaDetail](../resources/mfadetail.md)|**TODO: Add Description**|
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/signin.md:52]: TODO + the complex type.

File: api-reference/beta/resources/signin.md [hunk 5]
```diff
@@ -34,35 +34,51 @@ Provides details about user or application sign-in activity in your directory. Y
...
 |conditionalAccessStatus|conditionalAccessStatus| The status of the conditional access policy triggered. Possible values: `success`, `failure`, `notApplied`, or `unknownFutureValue`. Supports `$filter` (`eq` operator only).|
 |correlationId|String|The identifier that's sent from the client when sign-in is initiated. This is used for troubleshooting the corresponding sign-in activity when calling for support. Supports `$filter` (`eq` operator only).|
 |createdDateTime|DateTimeOffset|The date and time the sign-in was initiated. The Timestamp type is always in UTC time. For example, midnight UTC on Jan 1, 2014 is `2014-01-01T00:00:00Z`. Supports `$orderby` and `$filter` (`eq`, `le`, and `ge` operators only).|
+|crossTenantAccessType|signInAccessType|Describes the type of cross tenant access used by to access the resource in the sign-in event. Possible values are: `none`, `b2bCollaboration`, `b2bDirectConnect`, `microsoftSupport`, `serviceProvider`, `unknownFutureValue`. If the sign-in event did not cross tenant boundaries, the value is `none`.|
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/signin.md:43]: Incomplete sentence in description?

File: api-reference/beta/resources/signin.md [hunk 6]
```diff
@@ -34,35 +34,51 @@ Provides details about user or application sign-in activity in your directory. Y
...
 |deviceDetail|[deviceDetail](devicedetail.md)|The device information from where the sign-in occurred. Includes information such as deviceId, OS, and browser. Supports `$filter` (`eq` and `startsWith` operators only) on **browser** and **operatingSytem** properties.|
+|flaggedForReview|Boolean|During a failed sign in, a user may click a button in the Azure AD UX to mark the failed event for tenant admins. If a user clicked the button to flag a sign-in event, this value is true.|
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/signin.md:45]: Suggestion only if you meant the Portal and not another UI. If another UI, please indicate so.
```suggestion
|flaggedForReview|Boolean|During a failed sign in, a user may click a button in the Azure Portal to mark the failed event for tenant admins. If a user clicked the button to flag a sign-in event, this value is `true`.|
```

File: api-reference/beta/resources/signin.md [hunk 7]
```diff
@@ -34,35 +34,51 @@ Provides details about user or application sign-in activity in your directory. Y
...
+|homeTenantId|String|The tenant identifier of the user initiating the sign-in. Not applicable in Managed Identity or Service Principal sign-ins.|
 |id|String|The identifier representing the sign-in activity. Supports `$filter` (`eq` operator only).|
 |ipAddress|String|The IP address of the client from where the sign-in occurred. Supports `$filter` (`eq` and `startsWith` operators only).|
-|isInteractive|Boolean|Indicates whether a sign-in is interactive or not.|
+|isInteractive|Boolean|Indicates whether a user sign-in is interactive or not. In interactive sign-ins, the user provides an authentication factor to Azure AD. These factors include passwords, responses to MFA challenges, biometric factors, or QR codes that a user provides to Azure AD or to an associated app. In non-interactive sign-ins, the user does not provide an authentication factor; instead, the client app uses a token or code to authenticate or access a resource on behalf of a user. Non-interactive sign ins are commonly used for a client to sign in on a user's behalf in a process transparent to the user.|
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/signin.md:49]:
```suggestion
|isInteractive|Boolean|Indicates whether a user sign-in is interactive. In interactive sign-ins, the user provides an authentication factor to Azure AD. These factors include passwords, responses to MFA challenges, biometric factors, or QR codes that a user provides to Azure AD or to an associated app. In non-interactive sign-ins, the user does not provide an authentication factor; instead, the client app uses a token or code to authenticate or access a resource on behalf of a user. Non-interactive sign ins are commonly used for a client to sign in on a user's behalf in a process transparent to the user.|
```

File: api-reference/beta/resources/signin.md [hunk 8]
```diff
@@ -34,35 +34,51 @@ Provides details about user or application sign-in activity in your directory. Y
...
 |status|[signInStatus](signinstatus.md)|The sign-in status. Includes the error code and description of the error (in case of a sign-in failure). Supports `$filter` (`eq` operator only) on **errorCode** property.|
 |tokenIssuerName|String|The name of the identity provider. For example, `sts.microsoft.com`. Supports `$filter` (`eq` operator only).|
 |tokenIssuerType|tokenIssuerType|The type of identity provider. Possible values: `AzureAD`, `ADFederationServices`, or `UnknownFutureValue`.|
 |userAgent|String|The user agent information related to sign-in. Supports `$filter` (`eq` and `startsWith` operators only).|
 |userDisplayName|String|The display name of the user. Supports `$filter` (`eq` and `startsWith` operators only).|
 |userId|String|The identifier of the user. Supports `$filter` (`eq` operator only).|
 |userPrincipalName|String|The UPN of the user. Supports `$filter` (`eq` and `startsWith` operators only).|
+|userType|signInUserType|Identifies whether the user is a member or guest in the tenant that performed the sign in. Possible values are: `member`, `guest`, `unknownFutureValue`.|
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/signin.md:80]: missing Enum declaration for **signInUserType**.

File: changelog/Microsoft.AAD.Reporting.json [hunk 9]
```diff
@@ -1103,6 +1103,104 @@
...
+              "ChangeType": "Deletion",
+              "Description": "Removed the `unknownFutureValue` member from the **conditionalAccessRule** enumeration.",
+              "Target": "conditionalAccessRule"
+          }
+      ],
+      "Id": "2443cc12-cf8c-11eb-995f-5903f88189f5",
+      "Cloud": "prd",
+      "Version": "beta",
+      "CreatedDateTime": "2021-06-17T16:50:37.572Z",
+      "WorkloadArea": "Reports and audit",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.AAD.Reporting.json:1202]: Not aware if there's been a change from **Reports**.
```suggestion
      "WorkloadArea": "Reports",
```

File: changelog/Microsoft.AAD.Reporting.json [hunk 10]
```diff
@@ -1103,6 +1103,104 @@
...
+              "ChangeType": "Addition",
+              "Description": "Added the **isTenantRestricted** property to the [signIn](https://docs.microsoft.com/en-us/graph/api/resources/signIn?view=graph-rest-beta) resource.",
+              "Target": "signIn"
+          },
+          {
+              "Id": "244d59c2-cf8c-11eb-995f-5903f88189f5",
+              "ApiChange": "Property",
+              "ChangedApiName": "privateLinkDetails",
+              "ChangeType": "Addition",
+              "Description": "Added the **privateLinkDetails** property to the [signIn](https://docs.microsoft.com/en-us/graph/api/resources/signIn?view=graph-rest-beta) resource.",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.AAD.Reporting.json:1138]: - Seems not all properties added are listed here. Should include even those that were already in the schema but just missing from the live docs.
- Same GUID for all objects in the same ChangeList.
- Add new ChangeList in LIFO order.

File: api-reference/beta/api/signin-get.md (hunk 2 above)
Comment (besiler) [reply api-reference/beta/api/signin-get.md:113]: I think an empty array is right, but I'll check with Abbe to make sure

File: api-reference/beta/resources/signin.md (hunk 1 above)
Comment (besiler) [reply api-reference/beta/resources/signin.md:50]: Ack on this - can I edit them in the "Files Changed" tab in the PR or do I have to edit locally and push a new PR to fix it?

File: api-reference/beta/api/signin-get.md [hunk 11]
```diff
@@ -110,16 +110,32 @@ Content-type: application/json
...
   "conditionalAccessStatus": "notApplied",
   "originalRequestId": "66ea54eb-blah-4ee5-be62-ff5a759b0100",
+  "flaggedForReview": false,
+  "homeTenantId": "16645f6f-ad99-48b4-b1eb-193257b0dfdd",
   "isInteractive": true,
   "tokenIssuerName": "",
   "tokenIssuerType": "AzureAD",
+  "isTenantRestricted": false,
   "processingTimeInMilliseconds": 541,
+  "privateLinkDetails": {
```
[the remark this answers is not in the document]
Comment (besiler) [reply api-reference/beta/api/signin-get.md:133]: Should we prioritize making the example look "real" or do we just not need any conflicts? I can pull a real example, but then some properties will not have interesting values. Is it better for the example to have more explanatory value (but maybe look less real) as long as we don't have any true contradictions? 

Let me know what your guidance is and I'll adjust accordingly

File: api-reference/beta/api/signin-list.md [hunk 12]
```diff
@@ -111,15 +111,31 @@ Content-type: application/json
       "appId": "de8bc8b5-5555-6666-a8ad-b748da725064",
       "appDisplayName": "Graph explorer",
       "authenticationRequirement": "multiFactorAuthentication",
+      "authenticationRequirementPolicies":
```
[the remark this answers is not in the document]
Comment (besiler) [reply api-reference/beta/api/signin-list.md:114]: Ack - can regenerate an example, but we might lose some explanatory value. I'll do whatever you recommend as a best practice

File: api-reference/beta/resources/privatelinkdetails.md [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+doc_type: resourcePageType
+---
+
+# privateLinkDetails resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Provides details about the Azure Private Link associated with a sign-in event.
```
[the remark this answers is not in the document]
Comment (besiler) [reply api-reference/beta/resources/privatelinkdetails.md:16]: In docs.microsoft.com, not the Graph docs? Happy to do so

File: api-reference/beta/resources/signin.md [hunk 14]
```diff
@@ -34,35 +34,51 @@ Provides details about user or application sign-in activity in your directory. Y
...
+|isTenantRestricted|Boolean|**TODO: Add Description**|
 |location|[signInLocation](signinlocation.md)|The city, state, and 2 letter country code from where the sign-in occurred. Supports `$filter` (`eq` and `startsWith` operators only) on **city**, **state**, and **countryOrRegion** properties.|
+|mfaDetail|[mfaDetail](../resources/mfadetail.md)|**TODO: Add Description**|
 |networkLocationDetails|[networkLocationDetail](networklocationdetail.md) collection|The network location details including the type of network used and its names.|
 |originalRequestId|String|The request identifier of the first request in the authentication sequence. Supports `$filter` (`eq` operator only).|
+|privateLinkDetails|[privateLinkDetails](../resources/privatelinkdetails.md)|Contains information about the Azure AD Private Link policy associated with the sign-in event.|
```
[the remark this answers is not in the document]
Comment (besiler) [reply api-reference/beta/resources/signin.md:55]: Ack - asked the team for the best doc on Private Link Policy

File: api-reference/beta/resources/signin.md (hunk 8 above)
Comment (besiler) [reply api-reference/beta/resources/signin.md:80]: I'll add them. Enums are as follows:

o	interactiveUser 
o	nonInteractiveUser 
o	servicePrincipal 
o	managedIdentity

File: changelog/Microsoft.AAD.Reporting.json (hunk 10 above)
Comment (besiler) [reply changelog/Microsoft.AAD.Reporting.json:1138]: OK - if I have add something like this for the properties that weren't already in the docs, is that sufficient?

              "ApiChange": "Property",
              "ChangedApiName": "[name of API file]",
              "ChangeType": "Addition",
              "Description": "Added the **[property name]** property to the [signIn](https://docs.microsoft.com/en-us/graph/api/resources/signIn?view=graph-rest-beta) resource.",

Comment (besiler) [reply changelog/Microsoft.AAD.Reporting.json:1138]: Also, does LIFO mean adding the new changes at the top or bottom of the object?

Comment (FaithOmbongi) [reply changelog/Microsoft.AAD.Reporting.json:1138]: New changes at the start of the changelog doc.

File: api-reference/beta/resources/privatelinkdetails.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (FaithOmbongi) [reply api-reference/beta/resources/privatelinkdetails.md:16]: If it's in the Azure docs, feel free to link to it.

File: api-reference/beta/api/signin-get.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (FaithOmbongi) [reply api-reference/beta/api/signin-get.md:133]: Prioritize the example looking real. Feel free to remove properties that don't add value to the response, e.g. those with `null` values. Doesn't make sense to have half the response object having `null `values. If you remove uninteresting properties, there should be a truncation disclaimer:

>**Note:** The response object shown here might be shortened for readability.

Example: https://docs.microsoft.com/en-us/graph/api/user-list?view=graph-rest-1.0&tabs=http#response-1. 

Also, feel free to add additional examples if you wish to illustrate another context. Example: https://docs.microsoft.com/en-us/graph/api/accessreviewscheduledefinition-post?view=graph-rest-beta&tabs=http#example-2-create-an-access-review-on-all-teams-with-inactive-guest-users

File: api-reference/beta/api/signin-list.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (FaithOmbongi) [reply api-reference/beta/api/signin-list.md:114]: See previous response

File: api-reference/beta/resources/signin.md [hunk 15]
```diff
@@ -34,35 +34,51 @@ Provides details about user or application sign-in activity in your directory. Y
 |authenticationMethodsUsed|String collection|The authentication methods used. Possible values: `SMS`, `Authenticator App`, `App Verification code`, `Password`, `FIDO`, `PTA`, or `PHS`.|
 |authenticationProcessingDetails|[keyValue](keyvalue.md) collection|Additional authentication processing details, such as the agent name in case of PTA/PHS or Server/farm name in case of federated authentication.|
 |authenticationRequirement | String | This holds the highest level of authentication needed through all the sign-in steps, for sign-in to succeed. Supports `$filter` (`eq` and `startsWith` operators only).|
+|authenticationRequirementPolicies|[authenticationRequirementPolicy](../resources/authenticationrequirementpolicy.md) collection|Sources of authentication requirement, such as conditional access, per-user MFA, identity protection, and security defaults.|
+|autonomousSystemNumber|Int32|The Autonomous System Number (ASN) of the network used by the actor in the sign-in event.|
```
Comment (dkershaw10) [inline api-reference/beta/resources/signin.md:38]: Remove "in the sign-in event".  Same goes for crossTenantAccessType.

File: api-reference/beta/resources/signin.md (hunk 1 above)
Comment (dkershaw10) [reply api-reference/beta/resources/signin.md:50]: You're best off editing locally and pushing a new PR.  Address all the comments in one shot and push a new PR.

**Pull request closed without merging**